### DISCUSSION TITLE
Column Selection in Table Block and Tables No Longer Functions

### DISCUSSION THREAD

**Discussion by stellarfinn** [Budibase/budibase#11963] [Help] [answered]
At some point recently, the Table Block and Table components ability to select columns to be displayed has stopped working. 

When I select more than a single column, it shows all the columns in the table. 

I think this is in response to a recent update as it was working great before showing only a handful of columns out of dozens. 

Any thoughts on the cause or how to fix it?

See example below: 

Only selected two columns, but all columns displaying. 

<img width="1451" alt="image" src="https://github.com/Budibase/budibase/assets/99847728/13f19054-cf3c-4d72-b2ba-61c2667b1fcd">

**Comment by ConorWebb96** [comment]
Hey @stellarfinn,

What version of Budibase are you running? 

If you click the save button in the screenshot does it not change what the table displays? Initially, when configuring this setting no columns will be shown (This will cause all columns to be displayed as nothing has been specified). What happens if you click the button `add all columns` and then go through and delete the ones you don't wish to be displayed and then click the save button?

This should hopefully alter the columns which will be visible on the table.

I hope this helps.

**Reply by melohagan** [reply, marked as answer]
I think this can also happen if you are hiding some fields of the underlying data source in the **Data** section. If you click `Show all` columns in the **Data** section for the table being used, then the column order may be respected again

**Reply by stellarfinn** [reply]
This is the case. Reverting back to show all columns in the Data section fixed it!
